plugin/cancel/cancel.py

```diff
@@ -1,20 +1,70 @@
 # to disable command cancellation, comment out this entire file.
 # you may also wish to adjust the commands in misc/cancel.talon.
-from talon import actions, speech_system
+
+import time
+
+from talon import Context, Module, actions, speech_system
+from talon.grammar import Phrase
 
 # To change the phrase used to cancel commands, you must also adjust misc/cancel.talon
 cancel_phrase = "cancel cancel".split()
 
+mod = Module()
+ctx = Context()
+
+ts_threshold: float = 0
+
+
+@ctx.action_class("speech")
+class SpeechActions:
+    # When Talon wakes we set the timestamp threshold. On the next command we
+    # will compare the phrase timestamp to the threshold and cancel any phrase
+    # started before wakeup. This is to prevent speech said before wake-up to
+    # be interpreted as a command if the user wakes Talon using a noise or
+    # keypress.
+    def enable():
```

Comment:
We probably need to ensure that this coexists well with 

https://github.com/talonhub/community/blob/main/plugin/listening_timeout/listening_timeout.py

Which also overrides the enable() with the same specificity at the moment

Reply:
True. I didn't know about that feature. Do you mind testing it? We could of course hide this new behavior behind a tag or setting if we want to.

Reply:
False alarm, seems to work as-is for me. Nice improvement :)